### PR TITLE
[ES|QL] Document return value for locate in case substring is not found

### PR DESCRIPTION
**Pull request by ioanatia**
I wanted to use the `LOCATE` function for a test and noticed that we don't document the return value in case the substring is missing.

### CODE REVIEW

**Review by nik9000 — APPROVED** [review]

**Review by costin — CHANGES_REQUESTED** [review]
Please point out that the return position starts from 1 (not 0 as most programmers assume).

**Review by costin — CHANGES_REQUESTED** [review]
Please point out that the return position starts from 1 (not 0 as most programmers assume).

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/Locate.java [hunk 1]
```diff
@@ -46,11 +46,10 @@ public class Locate extends EsqlScalarFunction implements OptionalArgument {
     private final Expression substr;
     private final Expression start;
 
-    @FunctionInfo(
-        returnType = "integer",
-        description = "Returns an integer that indicates the position of a keyword substring within another string.",
-        examples = @Example(file = "string", tag = "locate")
-    )
+    @FunctionInfo(returnType = "integer", description = """
+        Returns an integer that indicates the position of a keyword substring within another string.
```
Comment (costin) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/Locate.java:50]: Returns an integer (starting from `1`) that indicates the position of a keyword substring within another string.
Returns `0` if the substring is not found.

**Review by costin — APPROVED** [review]
The text can be slightly rephrased to be more compact.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/Locate.java (hunk 1 above)
Comment (ioanatia) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/Locate.java:50]: Many developers are not used with string positions starting from 1. This is why I kind of like to have that as a separate note. If you feel very strongly about I can change this again, but for now I will merge the PR as it is.

**Pull request merged**